pygeoapi/api.py

```diff
@@ -3435,19 +3430,17 @@ def execute_process(self, request: Union[APIRequest, Any],
         url = f"{self.base_url}/jobs/{job_id}"
 
         headers['Location'] = url
-
-        is_async = data.get('mode', 'auto') == 'async'
-        if is_async:
-            LOGGER.debug('Asynchronous request mode detected')
-
-        if is_async and not self.manager.is_async:
-            LOGGER.debug('async manager not configured/enabled')
-            is_async = False
-
+        try:
+            execution_mode = RequestedProcessExecutionMode(
+                request.headers.get("Prefer"))
+        except ValueError:
+            execution_mode = None
         try:
             LOGGER.debug('Executing process')
-            mime_type, outputs, status = self.manager.execute_process(
-                process, job_id, data_dict, is_async)
+            execution_result = self.manager.execute_process(
+                process, job_id, data_dict, execution_mode=execution_mode)
+            mime_type, outputs, status, additional_headers = execution_result
+            headers.update(additional_headers or {})
```

Comment:
We now pass the requested execution mode to the process manager and let it decide whether to honor the request or not, based on its own configuration and that of the process being executed